### ISSUE TITLE
Fix job campaign in CMSMonitoring

### ISSUE THREAD

**Issue by khurtado** [dmwm/WMCore#11643]
**Impact of the new feature**
We need to propagate the job campaign in the CMS monitoring, preferentially as a new classad. 

**Is your feature request related to a problem? Please describe.**
Job campaign needs to be propagated for CMS monitoring

**Describe the solution you'd like**
Create a classad with the job campaign name. However, we don't seem to propagate this information in the job pickle, meaning the agent does not have this information as of now. We need to implement this feature at the job packet level before we can create the classad.

**Describe alternatives you've considered**

**Additional context**
This was meant to be implemented for the agent cycle (July 11-15), but it will have to be postponed, since we don't have this information easily accesible in the agent at this point.

**Comment by khurtado** [comment]
@amaltaro Here is the ticket for this issue, please feel free to correct/expand as needed.

**Comment by amaltaro** [comment]
Thanks for creating this issue, @khurtado .

I just wanted to complement that:
a) we need to set the job campaign at the job creation level (pickled job description)
b) consume the job campaign information at the JobSubmitter component (and SimpleCondorPlugin)
c) define the new job classad (perhaps `CMS_Campaign`).

We will have to discuss with CMS / CMS Monitoring on how to deal with StepChain workflows, because those jobs are for most of the cases multi-campaign, so we will likely create a comma separated list of campaigns and someone will have to define how to consume this information in monitoring.

Kenyi, I would suggest to rename this issue though to something like "Provide campaign name in the job classad". The actual fix will then be on the HTCondor job monitoring.

**Comment by khurtado** [comment]
@amaltaro I knew I was forgetting something important (stepChain comment), thank you, Alan!

**Comment by amaltaro** [comment]
Kenyi, I just noticed this is also a duplication of https://github.com/dmwm/WMCore/issues/10914, sorry about that! I am closing this one as not planned as well, and will add 10914 to the Q3/2023 project board.
